plugins.go

```diff
@@ -414,24 +415,32 @@ func (p *Plugins) unlockPlugin(name string) {
 
 // Update updates the plugins
 func (p *Plugins) Update() {
-	plugins := p.PluginNamesNotSymlinked()
-	if len(plugins) == 0 {
-		return
-	}
-	packages, err := p.OutdatedPackages(plugins...)
-	WarnIfError(err)
-	if len(packages) > 0 {
-		action("heroku-cli: Updating plugins", "", func() {
-			for name, version := range packages {
-				p.lockPlugin(name)
-				WarnIfError(p.installPackages(name + "@" + version))
-				_, err := p.ParsePlugin(name)
+	action("heroku-cli: Updating plugins", "done", func() {
+		for _, plugin := range p.Plugins() {
+			if p.isPluginSymlinked(plugin.Name) {
+				return
+			}
+			Errf("\rheroku-cli: Updating %s...", plugin.Name)
+			p.lockPlugin(plugin.Name)
+			tag := plugin.Tag
+			if tag == "" {
+				tag = "latest"
+			}
+			tags, err := p.DistTags(plugin.Name)
+			if err != nil {
 				WarnIfError(err)
-				p.unlockPlugin(name)
```

Comment:
@dickeyxxx why did you remove this?  better yet, you could move update to its own function and defer the unlock so that it always is unlocked